### PR TITLE
Closes #62

### LINKED ISSUE #62: Create a dataset loader for Nagel
From http://sourceforge.net/projects/bionlp-corpora/files/ProteinResidue/

### PR DESCRIPTION
**Pull request by HallerPatrick**
If the following information is NOT present in the issue, please populate:

- **Name:** **Nagel**
- **Description:** A set of 100 abstracts annotated by Kevin Nagel with protein, residue, organism triples.
- **Paper:** [Paper](https://www.ebi.ac.uk/sites/ebi.ac.uk/files/shared/documents/phdtheses/kevin_nagel.pdf)
- **Data:** https://sourceforge.net/projects/bionlp-corpora/files/ProteinResidue/

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: biodatasets/nagel/nagel.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+  author    = {Kevin Nagel},
+  title     = {Automatic functional annotation of predicted active sites: combining PDB and literature mining},
+  journal   = {},
+  volume    = {},
+  year      = {2009},
+  url       = {https://www.ebi.ac.uk/sites/ebi.ac.uk/files/shared/documents/phdtheses/kevin_nagel.pdf},
+  doi       = {},
+  biburl    = {},
+  bibsource = {}
+}
```
Comment (WojciechKusa) [inline biodatasets/nagel/nagel.py:40]: This bibtex might be more suitable: 
```suggestion
@phdthesis{nagel2009automatic,
  title={Automatic functional annotation of predicted active sites: Combining PDB and literature mining},
  author={Nagel, Kevin},
  year={2009},
  school={University of Cambridge}
  url= {https://www.ebi.ac.uk/sites/ebi.ac.uk/files/shared/documents/phdtheses/kevin_nagel.pdf},
}
```

File: biodatasets/nagel/nagel.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+
+        urls = _URLS[_DATASETNAME]
```
Comment (WojciechKusa) [inline biodatasets/nagel/nagel.py:145]: style related, could you replace `urls` to singular as it looks like this dataset uses only one URL?

```suggestion
        url = _URLS[_DATASETNAME]
```

**Review by WojciechKusa — CHANGES_REQUESTED** [review]
Hi @HallerPatrick, thanks for the contribution! :tada: 

I was reviewing your code, and unittests fail for me. You are using `str.removesuffix` added in Python 3.9, whereas `biomedical` currently supports Python 3.6+, so it would be nice to refactor that part for backwards compatibility. 

```
Traceback (most recent call last):
  File "/biomedical/tests/test_bigbio.py", line 176, in setUp
    self.dataset_source = datasets.load_dataset(
  File "/bigscience-biomedical/lib/python3.8/site-packages/datasets/load.py", line 1702, in load_dataset
    builder_instance.download_and_prepare(
  File "/bigscience-biomedical/lib/python3.8/site-packages/datasets/builder.py", line 594, in download_and_prepare
    self._download_and_prepare(
  File "/bigscience-biomedical/lib/python3.8/site-packages/datasets/builder.py", line 683, in _download_and_prepare
    self._prepare_split(split_generator, **prepare_split_kwargs)
  File "/bigscience-biomedical/lib/python3.8/site-packages/datasets/builder.py", line 1073, in _prepare_split
    for key, record in utils.tqdm(
  File "/bigscience-biomedical/lib/python3.8/site-packages/tqdm/std.py", line 1180, in __iter__
    for obj in iterable:
  File "/.cache/huggingface/modules/datasets_modules/datasets/nagel/871e024980006ad7b298ceac715797f818a5de147ffff85399e5b9115bf55fce/nagel.py", line 225, in _generate_examples
    pmid = int(file.name.removesuffix(".txt"))
AttributeError: 'str' object has no attribute 'removesuffix'
```

File: biodatasets/nagel/nagel.py (hunk 2 above)
Comment (hakunanatasha) [reply biodatasets/nagel/nagel.py]: fixed + with other references to url

**Comment by hakunanatasha** [comment]
@WojciechKusa I fixed the suffix line with a simple string split - all works on my end; great catch. I will merge it.

**Review by hakunanatasha — APPROVED** [review]

**Pull request merged**